### PR TITLE
#3831 - SINF restriction bridge mapping and rename to SINR

### PR DESCRIPTION
**Pull request by lewischen-aot**
- Changed restriction code `SINF` to `SINR`
- Updated the activity page accordion code to `SINR`
- Created a DB migration to change the code of the `SINF` to `SINR`.
- Renamed `SINF` to `SINR` in the existing `RestrictionCode` enum.
- Changed an existing E2E to ensure `SINF` will be mapped to `LGCY`. 
  - Replaced "TREE" with "SINF" in file "SFAS-TO-SIMS-2024MAR07-LEGACY-RESTRICTIONS.txt" to ensure `SINF` mapping to `LGCY` for this PR.

### Rollback evidence screenshot
image

Screenshot of the SINR code on the activity page accordion
image

Screenshot of the `sfas_student` table having two records after restrictions are imported
image

Screenshot of the `student_restrictions` table having only one record with restriction_id = 72 (LGCY)
image

Screenshot of the completed job for processing the SINR record on Bull Dashboard 
image

### CODE REVIEW

File: sources/packages/backend/apps/db-migrations/src/sql/Restrictions/Update-sin-restriction.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- Update action_type for the restriction code SINF
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/Restrictions/Update-sin-restriction.sql:1]: Minor. Update the `restriction_code`?

File: sources/packages/backend/libs/src-sql/sfas-restrictions/Bulk-insert-sfas-mapped-restrictions.sql [hunk 2]
```diff
@@ -22,6 +22,7 @@ FROM
         WHEN sfas_restrictions.code = 'A12' THEN '12'
         WHEN sfas_restrictions.code = 'AF1' THEN 'AF'
         WHEN sfas_restrictions.code = 'B2D' THEN 'B2'
+        WHEN sfas_restrictions.code = 'SINR' THEN 'LGCY'
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/src-sql/sfas-restrictions/Bulk-insert-sfas-mapped-restrictions.sql:25]: I do not believe that SINR will be sent from SFAS and I do not see a technical AC to add this condition.

Please let me know if I am missing something here?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-files/SFAS-TO-SIMS-2024MAR07-LEGACY-RESTRICTIONS.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 100AESTSFAS TO SIMS BRIDGE                    20211119142437 
 2001010101010JOHN           FOUR                                    19980324900041310 0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000                                                            
-40010101010100964351111TREE20180927   
+40010101010100964351111SINR20180927
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-files/SFAS-TO-SIMS-2024MAR07-LEGACY-RESTRICTIONS.txt:3]: The expectation is that, the restriction that needs to be mapped to LGCY is SINF not SINR.

image

**Comment by dheepak-aot** [comment]
Nice work @lewischen-aot. Please take a look at the comments.

**Review by andrepestana-aot — COMMENTED** [review]
I think there was a misunderstanding about the mapping (SINF should be imported from SFAS and treated as SINR in SIMS) and about the SINR if imported will be treated as LGCY as any other unknown restriction. Please check it.

File: sources/packages/backend/libs/src-sql/sfas-restrictions/Bulk-insert-sfas-mapped-restrictions.sql (hunk 2 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/src-sql/sfas-restrictions/Bulk-insert-sfas-mapped-restrictions.sql:25]: Please check `Bulk-insert-legacy-restrictions.sql`. This query will update with a `LGCY` if a restriction from SFAS is not found in SIMS.

**Review by andrepestana-aot — APPROVED** [review]
Looks good! Thanks for doing the changes.

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes. Looks good 👍

**Pull request merged**